cclib/method/volume.py

```diff
@@ -253,17 +282,43 @@ def electrondensity(ccdata, volume, mocoeffslist):
                     for j, yval in enumerate(y):
                         tmp = []
                         for zval in z:
-                            tmp.append(bfs[bs].amp(xval, yval, zval))
+                            tmp.append(pyamp(bfs, bs, xval, yval, zval))
                         data[i, j, :] = tmp
                 data *= mocoeff[bs]
                 wavefn += data
             density.data += wavefn ** 2
 
-    # TODO ROHF
-    if len(mocoeffslist) == 1:
-        density.data *= 2.0
-
     return density
 
+def electrondensity(ccdata, volume, mocoeffslist):
+    """Calculate the magnitude of the total electron density at every point in a volume.
 
+    Inputs:
+        ccdata -- ccData object
+        volume -- Volume object (will not be altered)
+        mocoeffslist -- list of molecular orbital to calculate electron density from;
+                        i.e. [ccdata.mocoeffs[0][1:2]]
+    
+    Output:
+        Volume object with wavefunction at each grid point stored in data attribute
+
+    Attributes:
+        coords -- the coordinates of the atoms
+        mocoeffs -- mocoeffs for all of the occupied eigenvalues
+        gbasis -- gbasis from a parser object
+        volume -- a template Volume object (will not be altered)
+
+    Note: mocoeffs is a list of NumPy arrays. The list will be of length 1
+          for restricted calculations, and length 2 for unrestricted.
+    """
+
+    if len(mocoeffslist) == 2:
+        return electrondensity_spin(ccdata, volume, [mocoeffslist[0]]) + electrondensity_spin(ccdata, volume, [mocoeffslist[1]])
+    else:
+        edens = electrondensity_spin(ccdata, volume, [mocoeffslist[0]])
+        edens.data *= 2
+        return edens

```

Comment:
This looks good for now but in the future this should be modified to take a `coeffslist` that would either be 2D or 1D and an occupation number list that scales the spin density by (2D case) 1*occno[i] or (1D case) 2*occno[i]. This would require changes in `electrondensity_spin` but it would allow us to use NOs as well as MOs. No necessary for this PR, but I thought I would add a note.